File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/ThemeUtil.java
package org.dolphinemu.dolphinemu.utils;

import android.os.Build;

import androidx.appcompat.app.AppCompatDelegate;

import org.dolphinemu.dolphinemu.features.settings.model.Setting;
import org.dolphinemu.dolphinemu.features.settings.model.Settings;
import org.dolphinemu.dolphinemu.features.settings.utils.SettingsFile;

public class ThemeUtil
{
  public static void applyTheme()
  {
    Settings settings = new Settings();

    try
    {
      settings.loadSettings(null);
      applyTheme(settings);
    }
    catch (Exception e)
    {
      applyTheme(2);
    }
  }

  private static void applyTheme(int designValue)
  {
    switch (designValue)
    {
      case 0:
        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
        break;
      case 1:
        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
        break;
      case 2:
        AppCompatDelegate.setDefaultNightMode(android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q ?
        AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM :
        AppCompatDelegate.MODE_NIGHT_AUTO_BATTERY);
        break;
    }
  }

  public static void applyTheme(Settings settings)
  {
    Setting design = settings.getSection(Settings.SECTION_INI_INTERFACE).getSetting((SettingsFile.KEY_DESIGN));
    if (design == null)
    {
      AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
      return;
    }

    applyTheme(Integer.parseInt(design.getValueAsString()));
  }
}

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/ThemeUtil.kt
package org.dolphinemu.dolphinemu.utils

import android.os.Build
import androidx.appcompat.app.AppCompatDelegate
import org.dolphinemu.dolphinemu.features.settings.model.Settings
import org.dolphinemu.dolphinemu.features.settings.utils.SettingsFile

object ThemeUtil {
    fun applyTheme() {
        val settings = Settings()

        try {
            settings.loadSettings(null)
            applyTheme(settings)
        } catch (e: Exception) {
            applyTheme(2)
        }
    }

    private fun applyTheme(designValue: Int) {
        when (designValue) {
            0 -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
            1 -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
            2 -> AppCompatDelegate.setDefaultNightMode(if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM else AppCompatDelegate.MODE_NIGHT_AUTO_BATTERY)
        }
    }

    fun applyTheme(settings: Settings) {
        val design = settings.getSection(Settings.SECTION_INI_INTERFACE)
            .getSetting((SettingsFile.KEY_DESIGN))
        if (design == null) {
            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
            return
        }

        applyTheme(design.valueAsString.toInt())
    }
}
